packages/libs/src/sdk/services/Solana/programs/ClaimableTokens.ts

```diff
@@ -0,0 +1,196 @@
+import { wAUDIO } from '@audius/fixed-decimal'
+import { ClaimableTokensProgram } from '@audius/spl'
+import {
+  TransactionMessage,
+  VersionedTransaction,
+  Secp256k1Program,
+  PublicKey,
+  Connection
+} from '@solana/web3.js'
+import { parseParams } from '../../../utils/parseParams'
+import type { Mint, SolanaService } from '../types'
+
+import {
+  type GetOrCreateUserBankRequest,
+  GetOrCreateUserBankSchema,
+  type DeriveUserBankRequest,
+  type CreateTransferRequest,
+  CreateTransferSchema,
+  type CreateSecpRequest,
+  CreateSecpSchema,
+  ClaimableTokensConfigInternal
+} from './types'
+
+import * as runtime from '../../../api/generated/default/runtime'
+
+export class ClaimableTokens {
+  /** The program ID of the ClaimableTokensProgram instance. */
+  private readonly programId: PublicKey
+  /** Connection to interact with the Solana RPC. */
+  private readonly connection: Connection
+  /** Map from token mint name to public key address. */
+  private readonly mints: Record<Mint, PublicKey>
+  /** Map from token mint name to derived user bank authority. */
+  private readonly authorities: Record<Mint, PublicKey>
+
+  constructor(
+    config: ClaimableTokensConfigInternal,
+    private solana: SolanaService
+  ) {
+    this.programId = config.programId
+    this.connection = config.connection
+    this.mints = config.mints
+    this.authorities = {
+      wAUDIO: ClaimableTokensProgram.deriveAuthority({
```

Comment:
Any concerns about deriving these in the ctor vs. just passing in constants?

Reply:
I don't think so. I think deriving means less chance of error and the cost doesn't seem significant.